misc/pytorch_toolkit/breast_ultrasound_simulation/src/solve.py

```diff
@@ -0,0 +1,358 @@
+
+                # Cumulative loss for analysis
+                L1_epoch += self.reconstruction_loss.item()
+                AdvG_epoch += self.generator_adversarial_loss.item()
+                AdvD_epoch += self.adversarial_loss.item()
+                # print(time.time())
+                # Python Logging
+                print('\rStep [{}/{}], L1: {:.4f}, AccD: {:.4f}, AdvD: {:.4f}, AdvG: {:.4f} Time: {:.1f}sec'.format((i_batch + 1),
+                                                                                                                    len(
+                                                                                                                        self.train_data),
+                                                                                                                    L1_epoch /
+                                                                                                                    (i_batch + 1),
+                                                                                                                    DisAcc_epoch /
+                                                                                                                    (i_batch + 1),
+                                                                                                                    AdvD_epoch /
+                                                                                                                    (i_batch + 1),
+                                                                                                                    AdvG_epoch /
+                                                                                                                    (i_batch + 1),
+                                                                                                                    time.time() - start_time),
+                      end=" ")
+
+                # tensorboard logging
+                if(self.log):
+                    if(i_batch % args.log_step == 0):
+                        self.writer.add_scalar('Loss/Discriminator',
+                                               self.adversarial_loss.item(),
+                                               i_batch + len(self.train_data) * epoch)
+                        self.writer.add_scalar('Loss/Generator',
+                                               self.generator_adversarial_loss.item(),
+                                               i_batch + len(self.train_data) * epoch)
+                        self.writer.add_scalar('Loss/Reconstruction',
+                                               self.reconstruction_loss.item(),
+                                               i_batch + len(self.train_data) * epoch)
+                        self.writer.add_scalar('LR', get_lr(
+                            self.g_opt), i_batch + len(self.train_data) * epoch)
+
+                    if(i_batch == 0 and epoch % self.args.vis_step == 0):
```

Comment:
```suggestion
                    if i_batch == 0 and epoch % self.args.vis_step == 0 :
```
No need to add parenthesis in if condition